src/Gaufrette/Adapter/AwsS3.php

```diff
@@ -111,10 +111,12 @@ public function rename($sourceKey, $targetKey)
 
         try {
             $this->service->copyObject($options);
-            return true;
         } catch (\Exception $e) {
             return false;
         }
+        $this->delete($sourceKey);
```

Comment:
@shieldo not sure. it reads like that with rename either the operation fails and only the initial source will stay as is or the new destination will be created while the source will be gone, which sounds like a atomic rename.

A copy+rename in 2 operations cannot be atomic using this PHP-APIs IIRC.